pkg/schedule/schedulers/hot_region.go

```diff
@@ -365,17 +386,16 @@ func (h *hotScheduler) balanceHotReadRegions(cluster sche.SchedulerCluster) []*o
 
 func (h *hotScheduler) balanceHotWriteRegions(cluster sche.SchedulerCluster) []*operator.Operator {
 	// prefer to balance by peer
-	s := h.r.Intn(100)
-	switch {
-	case s < int(schedulePeerPr*100):
+	// try to move peer
+	if h.r.Float64() < h.probabilityToMovePeer {
```

Comment:
what is the probability effect, I feel that the benefits are not obvious. and in the case of 5 replicas + 2 tiflash, the leader may not have many chances to be scheduled?

Reply:
In the past, the probability of scheduling was based on the assumption of three replicas to calculate 2/3 = 0.66, it is not accurate for the probability under multiple replicas.